pkg/metallb/frr_test.go

```diff
@@ -0,0 +1,510 @@
+package metallb
+
+import (
+	"fmt"
+	"testing"
+	"time"
+
+	"github.com/openshift-kni/eco-goinfra/pkg/clients"
+	"github.com/openshift-kni/eco-goinfra/pkg/schemes/metallb/frrtypes"
+	"github.com/stretchr/testify/assert"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+)
+
+var (
+	frrTestSchemes                = []clients.SchemeAttacher{frrtypes.AddToScheme}
+	defaultFrrConfigurationName   = "frrconfiguration"
+	defaultFrrConfigurationNsName = "test-namespace"
+	apiClient                     *clients.Settings
+)
+
+func TestNewFrrConfigurationBuilder(t *testing.T) {
+	generateFrrConfiguration := NewFrrConfigurationBuilder
+
+	testCases := []struct {
+		runTimeClient *clients.Settings
+		name          string
+		namespace     string
+		expectedError string
+	}{
+		{
+			runTimeClient: apiClient,
+			name:          defaultFrrConfigurationName,
+			namespace:     defaultFrrConfigurationNsName,
+			expectedError: "",
+		},
+		{
+			runTimeClient: apiClient,
+			name:          "",
+			namespace:     defaultFrrConfigurationNsName,
+			expectedError: "frrConfiguration 'name' cannot be empty",
+		},
+		{
+			runTimeClient: apiClient,
+			name:          defaultFrrConfigurationName,
+			namespace:     "",
+			expectedError: "frrConfiguration 'nsname' cannot be empty",
+		},
+		{
+			runTimeClient: nil,
+			name:          defaultFrrConfigurationName,
+			namespace:     defaultFrrConfigurationNsName,
+			expectedError: "",
+		},
+	}
+
+	for _, testCase := range testCases {
+		testSettings := clients.GetTestClients(clients.TestClientParams{SchemeAttachers: frrTestSchemes})
+		testFRRConfigurationBuilder := generateFrrConfiguration(testSettings, testCase.name, testCase.namespace)
+		assert.Equal(t, testCase.expectedError, testFRRConfigurationBuilder.errorMsg)
+		assert.NotNil(t, testFRRConfigurationBuilder.Definition)
+
+		if testCase.expectedError == "" {
+			assert.Equal(t, testCase.name, testFRRConfigurationBuilder.Definition.Name)
+			assert.Equal(t, testCase.namespace, testFRRConfigurationBuilder.Definition.Namespace)
+		}
+	}
+}
+
+func TestFrrConfigurationCreate(t *testing.T) {
+	testCases := []struct {
+		testFrrConfiguration *FrrConfigurationBuilder
+		expectedError        error
+	}{
+		{
+			testFrrConfiguration: buildValidFRRConfigurationBuilder(buildFrrConfigurationTestClientWithDummyObject()),
+			expectedError:        nil,
+		},
+		{
+			testFrrConfiguration: buildInValidFRRConfigurationBuilder(buildMetalLbTestClientWithDummyObject()),
```

Comment:
I think this is the wrong function for getting an object since it's for metallb not frrconfig

Reply:
Yes that fixed it. I also removed the typemeta.
Thanks